### PR TITLE
Recurring aa test

### PR DESCRIPTION
**Pull request by Mullefa**
@desbo slight modification to allow for finer grain control on setting complete flags for tests.

### CODE REVIEW

File: assets/javascripts/src/modules/abTests.es6 [hunk 1]
```diff
@@ -8,11 +8,9 @@ import { SET_AMOUNT } from 'src/actions';
 export function init() {
     const state = store.getState();
 
-    ophan.loaded.then(function (ophan) {
-        ophan.record({
-            abTestRegister: abTestData(state.data.abTests, false)
-        })
-    });
+    registerTestsWithOphan(state.data.abTests, initTestFlags);
```
Comment (desbo) [inline assets/javascripts/src/modules/abTests.es6]: I'm not sure this really needs the second parameter. At this point it's fine to just send `complete: false` for all of the active tests. Sending `complete: true` is down to each individual test, and might happen at any point in the app's lifecycle so it should be enough to just provide a helper function that fires the `true` event for a test.

File: assets/javascripts/src/modules/abTests.es6 [hunk 2]
```diff
@@ -21,27 +19,94 @@ export function init() {
...
+   return isAARecurringTest(test);
+}
+
+/**
+ * To be used with registerTestsWithOphan().
+ * Set a test's complete flag to true, apart from the AA recurring test which is ignored.
+ */
+function completeTestFlags(test) {
+    if (isAARecurringTest(test)) {
+        return null;
```
Comment (desbo) [inline assets/javascripts/src/modules/abTests.es6]: My idea for this test was to fire `true` straight away so we should see as many `falses` as there are `true`s

**Comment by Mullefa** [comment]
@desbo based on your comments I've removed the second parameter from `registerTestsWithOphan()`. Now, when the page loads, all test complete flags are set to false. I realised for the AA test this is enough, so I haven't bothered setting its flag to true at any point.

Regarding the helper function for setting a test complete flag to true, I thought it could be implemented as needed, but happy to do it as part of this pull request if you want?

**Comment by desbo** [comment]
> Now, when the page loads, all test complete flags are set to false. I realised for the AA test this is enough, so I haven't bothered setting its flag to true at any point.

The problem before was that we weren't seeing as many `true`s as there were events in GA. I would still have the A/A test fire a `true` event as well as `false`, but for this test it's fine to fire `true` as soon as the page loads. 

> Regarding the helper function for setting a test complete flag to true, I thought it could be implemented as needed, but happy to do it as part of this pull request if you want?

Because we only run one test a time, the code would be exactly the same as `registerTestsWithOphan` but with `complete` set to `true`. I would just update that function to take a boolean parameter for `complete`.  You then run that function with `false` at the start to register everything and  the test can call it with `true` later.

File: app/views/support/ABTest.scala [hunk 3]
```diff
@@ -162,6 +157,18 @@ object Test {
   }
 }
 
+/**
+ * The first time the recurring test was run, there was a big discrepancy between the data in Ophan and GA.
+ * This test fires an event to Ophan on page load; (independently) an event will also be fired to GA on page load.
+ * By comparing the event counts, we hope to debug the discrepancy.
+ */
```
Comment (philwills) [inline app/views/support/ABTest.scala:164]: 👍 for comments that convey rationale rather than implementation

**Review by desbo — APPROVED** [review]

**Pull request merged**